internal/providers/terraform/aws/rds_cluster.go

```diff
@@ -0,0 +1,213 @@
+package aws
+
+import (
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+	"strings"
+)
+
+func GetRDSClusterRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_rds_cluster",
+		RFunc: NewRDSCluster,
+	}
+}
+
+func NewRDSCluster(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	region := d.Get("region").String()
+
+	costComponents := make([]*schema.CostComponent, 0)
+
+	databaseEngineMode := "provisioned"
+	if d.Get("engine_mode").Exists() {
+		databaseEngineMode = strings.Title(d.Get("engine_mode").String())
+	}
+
+	databaseEngineStorageType := strPtr("Any")
+
+	var databaseEngine *string
+	switch d.Get("engine").String() {
+	case "aurora", "aurora-mysql":
+		databaseEngine = strPtr("Aurora MySQL")
+	case "aurora-postgresql":
+		databaseEngine = strPtr("Aurora PostgreSQL")
+		databaseEngineStorageType = databaseEngine
+	}
+
+	var auroraCapacityUnits *decimal.Decimal
+	if u != nil && u.Get("capacity_unit_hrs").Exists() {
+		auroraCapacityUnits = decimalPtr(decimal.NewFromInt(u.Get("capacity_unit_hrs").Int()))
+	}
+
+	if databaseEngineMode == "Serverless" {
+		costComponents = append(costComponents, &schema.CostComponent{
+			Name:           "Aurora serverless",
+			Unit:           "hours",
+			UnitMultiplier: 1,
+			HourlyQuantity: auroraCapacityUnits,
+			ProductFilter: &schema.ProductFilter{
+				VendorName:    strPtr("aws"),
+				Region:        strPtr(region),
+				Service:       strPtr("AmazonRDS"),
+				ProductFamily: strPtr(databaseEngineMode),
+				AttributeFilters: []*schema.AttributeFilter{
+					{Key: "databaseEngine", Value: databaseEngine},
+				},
+			},
+		})
+	}
+
+	costComponents = append(costComponents, auroraStorageCostComponent(region, u, databaseEngineStorageType)...)
+
+	backupStorageRetention := decimal.NewFromInt(d.Get("backup_retention_period").Int())
+	if backupStorageRetention.GreaterThan(decimal.NewFromInt(1)) {
+		snapShotStorageSizeGB := decimal.Zero
+		if u != nil && u.Get("backup_snapshot_size_gb").Exists() {
+			snapShotStorageSizeGB = decimal.NewFromFloat(u.Get("backup_snapshot_size_gb").Float())
+		}
+
+		totalBackupStorageGB := calculateBackupStorage(snapShotStorageSizeGB, backupStorageRetention)
+
+		if totalBackupStorageGB.GreaterThan(decimal.Zero) {
+			costComponents = append(costComponents, auroraBackupStorageCostComponent(region, totalBackupStorageGB, databaseEngineStorageType))
+		}
+	}
+
+	backtrackChangeRecords := decimal.Zero
+	if u != nil && u.Get("backtrack_change_records").Exists() {
+		backtrackChangeRecords = decimal.NewFromInt(u.Get("backtrack_change_records").Int())
+	}
+
+	backtrackPeriod := decimal.Zero
+	if u != nil && u.Get("backtrack_change_record_hrs").Exists() {
+		backtrackPeriod = decimal.NewFromInt(u.Get("backtrack_change_record_hrs").Int())
+	}
+
+	totalBacktrackChangeRecords := backtrackChangeRecords.Mul(backtrackPeriod)
```

Comment:
I'm unsure what we should use for the calculation here. The AWS calculator has the following:
* Average statements (per sec, hour, etc)
* Change records per statement
* Target backtrack window (hours)

The calculation is all these numbers multiplied together. This makes me lean towards having the following usage fields:
* average_statements_per_hr
* change_requests_per_statement
* backtrack_window_hrs